pkg/kubectl/cmd/util/factory.go

```diff
@@ -176,7 +177,25 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			if cfg.GroupVersion != nil {
 				cmdApiVersion = *cfg.GroupVersion
 			}
-
+			client, err := clients.ClientForVersion(&unversioned.GroupVersion{Version: "v1"})
+			if err != nil {
+				fmt.Printf("Failed to create client: %v", err)
+				return nil, nil
+			}
+			versions, gvks, err := GetThirdPartyGroupVersions(client.RESTClient)
```

Comment:
I don't want to make it a separate function, because then we'll have to add it to every single place a factory is used, I want this to work by default anywhere a `Builder` is used.

I'd be ok with a flag to disable, is that ok w/ you?